src/components/List/List.tsx

```diff
@@ -119,7 +129,7 @@ class List extends UIComponent<Extendable<ListProps>, ListState> {
     this.focusHandler = new ContainerFocusHandler(
       () => this.props.items.length,
       index => {
-        this.setState({ selectedItemIndex: index }, () => {
+        this.setState({ focusedItemIndex: index }, () => {
           const targetComponent = this.itemRefs[index] && this.itemRefs[index].current
           const targetDomNode = ReactDOM.findDOMNode(targetComponent) as any
```

Comment:
shouldn't `selectedIndex` be updated as well?

Reply:
no need for that, ```ContainerFocusHandler``` is just responsible for moving focus during navigation. When you actually select an item by clicking, it also forces onFocus and updates ```focusedItemIndex```. When you select an item by pressing Enter or Space, it means the focus is already on the element that you're going to select.